Sandbox/Sandbox.cpp

```diff
@@ -3,52 +3,78 @@
 #include <glm/gtc/matrix_transform.hpp>
 
 SandboxLayer::SandboxLayer()
-	: Layer("Sandbox")
+    : Layer("Sandbox")
 {}
 
 void SandboxLayer::onCreation()
 {
-	m_vertexBuffer = std::make_shared<Aquarius::VertexBuffer>(
-		m_triangleVertices.data(), m_triangleVertices.size() * sizeof(float));
-
-	m_indexBuffer = std::make_shared<Aquarius::IndexBuffer>(m_indices.data(), m_indices.size());
-
-	m_bufferLayout = std::make_shared<Aquarius::BufferLayout>(std::initializer_list<Aquarius::VertexElement>{
-		Aquarius::VertexElement(Aquarius::ShaderType::Float, 2, false),
-		Aquarius::VertexElement(Aquarius::ShaderType::Float, 3, false),
-		Aquarius::VertexElement(Aquarius::ShaderType::Float, 2, false)
-	});
-
-	m_vertexArray = std::make_shared<Aquarius::VertexArray>(m_vertexBuffer, m_indexBuffer, *m_bufferLayout);
-	m_vertexArray->activate();
-	Aquarius::Renderer::Init();
-	m_texture = std::make_shared<Aquarius::Texture>("Assets/container.jpg");
-	m_texture->bind(0);
+    m_ShaderProgram = std::make_shared<Aquarius::Shader>("Sandbox/Shaders/modifiableVertexShader.glsl", 
+                                                         "Sandbox/Shaders/modifiableFragmentShader.glsl");
+
+    m_vertexBuffer = std::make_shared<Aquarius::VertexBuffer>(
+        m_triangleVertices.data(), m_triangleVertices.size() * sizeof(float));
+
+    m_indexBuffer = std::make_shared<Aquarius::IndexBuffer>(m_indices.data(), m_indices.size());
+
+    m_bufferLayout = std::make_shared<Aquarius::BufferLayout>(std::initializer_list<Aquarius::VertexElement>{
+        Aquarius::VertexElement(Aquarius::ShaderType::Float, 2, false),
+        Aquarius::VertexElement(Aquarius::ShaderType::Float, 3, false),
+        Aquarius::VertexElement(Aquarius::ShaderType::Float, 2, false)
+    });
+
+    m_vertexArray = std::make_shared<Aquarius::VertexArray>(m_vertexBuffer, m_indexBuffer, *m_bufferLayout);
+    m_vertexArray->activate();
+
+    Aquarius::Application* app = Aquarius::Application::get();
+    Aquarius::Window* window = app->getWindow();
```

Comment:
This is satisfying to me for some reason haha, I find it cool how we have global access to the application class. Idk why everybody be hating on globals/singletons

Reply:
Yeah UE4 has a similar thing with a class called  GEngine